File: src/main/java/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationIsForIntegration.java
package no.novari.flyt.integration.validation.constraints;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = ReferencedConfigurationIsForIntegrationValidator.class)
public @interface ReferencedConfigurationIsForIntegration {

    String message() default "referenced configuration is not for this integration";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationIsForIntegration.kt
package no.novari.flyt.integration.validation.constraints

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [ReferencedConfigurationIsForIntegrationValidator::class])
annotation class ReferencedConfigurationIsForIntegration(
    val message: String = "referenced configuration is not for this integration",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
